### PR TITLE
Suggestion: a modest helper to reduce boilerplate

### PR DESCRIPTION
**Pull request by inopinatus**
I've been following the View Components journey and I was inevitably curious about the Primer components repo.  The instantiation boilerplate immediately struck me as verbose, certainly for own my personal taste, and my instinct was to immediately tuck that into a helper to improve legibility.

So I'm just sending this PR by way of suggestion, you may of course have other ideas or style preferences!

### CODE REVIEW

File: lib/primer/view_helper.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+# Provides a convenient shorthand for components.
+module Primer::ViewHelper
```
Comment (manuelpuyol) [inline lib/primer/view_helper.rb:4]: could you break it into

```rb
module Primer
  module ViewHelper
```
?

File: lib/primer/view_helper.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  Components = Hash.new do |hash, component|
+    hash[component] = "Primer::#{component.to_s.camelize}Component".constantize
+  end
+
+  def primer(component, **kwargs, &block)
+    render Components[component].new(**kwargs), &block
+  end
```
Comment (manuelpuyol) [inline lib/primer/view_helper.rb:11]: I'd like to see some error handling here 😄

File: lib/primer/view_helper.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+# Provides a convenient shorthand for components.
+module Primer::ViewHelper
+  Components = Hash.new do |hash, component|
+    hash[component] = "Primer::#{component.to_s.camelize}Component".constantize
```
Comment (manuelpuyol) [inline lib/primer/view_helper.rb:6]: for now I guess this should be fine, but I imagine we'll have some more complex namespaces in the future

**Comment by joelhawksley** [comment]
@inopinatus thanks for taking the time to make this suggestion via a PR! It's super helpful to have proposed code when discussing ideas likes this. I too share a distaste for how verbose `render` calls can look with ViewComponents.

However, we've generally avoided introducing a shorthand for rendering components, as we have several static analysis tools internally that rely on component class names being referenced directly. I'm hesitant to make this change now given that dependency.

File: lib/primer/view_helper.rb (hunk 1 above)
Comment (inopinatus) [reply lib/primer/view_helper.rb:4]: 👍

File: lib/primer/view_helper.rb (hunk 2 above)
Comment (inopinatus) [reply lib/primer/view_helper.rb:11]: Good thought. The most likely due to the helper is a NameError. Although my default handling would be to re-throw it.

**Comment by inopinatus** [comment]
@joelhawksley @manuelpuyol thanks for taking a look! I'll close this PR for now.

I am making a ViewComponent-based library for my own app's UI, so I fully intend to shamelessly borrow from whatever structures you land on.

**Pull request closed without merging**